### ISSUE TITLE
do not report the value type of map with a non-nullable value type

### ISSUE THREAD

**Issue by robstoll** [robstoll/atrium#285]
*Platform* (jvm, js, android): all
*Extension* (none, kotlin 1.3, jdk8): none

## Code related feature
```
expect(mapOf("a" to 1, "b" to null)).contains("b" to 2)
```
Results in the following report:
```
expect: {a=1, b=null}        (java.util.LinkedHashMap <1207769059>)
◆ contains, in any order: 
  ⚬ ▶ entry "b": null
      ◾ is instance of type: Int (kotlin.Int) -- Class: Integer (java.lang.Integer)
        » to be: 2        (kotlin.Int <451111351>)
```

The extra hint about the type makes only sense if the Map has a nullable value type. IF it doesn't it does not make sense. A user with a Reporter which reports all assertions would see this hint also for non-nullable value types where it is superfluous. Thus add an overload with a non-nullable value type which does not include this hint.

**Comment by robstoll** [comment]
Unfortunately, we cannot implement it due to this JS bug: https://youtrack.jetbrains.com/issue/KT-33294 
There is a workaround we already apply to `toBe` but IMO we can wait with this enhancement until the bug is fixed.

**Comment by jGleitz** [comment]
related: in https://github.com/robstoll/atrium-roadmap/issues/7, we are discussing to drop the instance check from `toBe` entirely. This would probably then also apply to this assertion as well.

**Comment by robstoll** [comment]
We have to implement contains for value not in terms of toBeNullIfNullGivenElse to achieve this but with toBe. Once https://github.com/robstoll/atrium/issues/298 is implemented, then this should be correct as well.

**Comment by robstoll** [comment]
implemented with 8476fa1a
